### ISSUE TITLE
Add an "Add Script" UI to the Create Module app in Setup

### ISSUE THREAD

**Issue by unsoluble** [foundryvtt/foundryvtt#11905]
### User Experience

A lot of end-user scripting that can't be handled by macros has been traditionally solved with world scripts. These certainly have fair uses, but it's often _better_ to have those scripts contained in a module, for easy troubleshooting and portability.

Now that we have a GUI tool in Setup for easily creating a local module, I think it'd be great if there could be an optional field in that config for directly adding a script to the package. This would _mostly_ eliminate the community support burden around the addition of scripts to world packages.

The UI here may take some hashing out, but I imagine it as something like:

[Checkbox, disabled by default] Add Script File [filename field, with file browser button]
Hint text explaining this a bit

**Comment by unsoluble** [comment]
Yeah, I feel your sentiment. The reason I think this would be helpful regardless is that, in practice, folks are simply not able to handle the task of modifying JSON without error.

We have an extremely thorough and explicit guide on the matter on the wiki, and are still met with failures daily. The vast majority of people adding world scripts aren’t _writing_ them, but are just pasting something they were given in #macro-polo — pasting into a single file with no arcane syntax requirements around the edges would help, is the crux of it.

**Comment by mxzf** [comment]
If this does get added, it would be ideal to implement an option for `styles` the same way.  Same idea really, it's not uncommon for users to ask for UI tweaks and get a CSS block that they need to paste somewhere in order to load it in.  As it stands the Custom CSS module ends up getting recommended for such things, but making it easier to set up a custom bespoke module with personal tweaks like that would be ideal.

**Comment by mclemente** [comment]
Bumping this issue to suggest moving this idea from the Setup and Module Creation to the actual world using V13's Macro sidebar.

A new "World Script" macro type, essentially a Script macro that can't be placed on the macro hotbar and runs on world load. Alternatively, just a checkbox for Script macros to run them on world load.

The Advanced Macros module has a similar feature but it is limited to running at the `setup` hook at the earliest since `game.macros` isn't available before that, so it still can't cover all that a World Script can normally do.
